subsystem/core/src/main/java/org/wildfly/extension/camel/service/CamelEndpointDeployerService.java

```diff
@@ -0,0 +1,480 @@
+                endpointSecurityConstraint.setTransportGuaranteeType(
+                        transportGuaranteeType(uri, mainSecurityConstraint.getTransportGuaranteeType()));
+                result.add(endpointSecurityConstraint);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * This method can simplified substantially, once https://github.com/undertow-io/undertow/pull/642 reaches us.
+     * Currently, the method is just an adjusted copy of {@link DeploymentInfo#clone()}.
+     *
+     * @param securityConstraints
+     * @param servletInfo
+     * @param deploymentName
+     * @param contextPath
+     */
+    static DeploymentInfo adaptDeploymentInfo(DeploymentInfo src, URI uri, ServletInfo servletInfo) {
+        final String contextPath = uri.getPath();
+        final String deploymentName = uri.getPath();
+
+        final DeploymentInfo info = new DeploymentInfo() //
+                .setClassLoader(src.getClassLoader()) //
+                .setContextPath(contextPath) //
+                .setResourceManager(src.getResourceManager()) //
+                .setMajorVersion(src.getMajorVersion()) //
+                .setMinorVersion(src.getMinorVersion()) //
+                .setDeploymentName(deploymentName) //
+                .setClassIntrospecter(src.getClassIntrospecter());
+
+        info.addServlet(servletInfo);
+
+        for (Map.Entry<String, FilterInfo> e : src.getFilters().entrySet()) {
+            info.addFilter(e.getValue().clone());
+        }
+        info.setDisplayName(src.getDisplayName());
+        for (FilterMappingInfo fmi : src.getFilterMappings()) {
+            switch (fmi.getMappingType()) {
+            case URL:
+                info.addFilterUrlMapping(fmi.getFilterName(), fmi.getMapping(), fmi.getDispatcher());
+                break;
+            case SERVLET:
+                info.addFilterServletNameMapping(fmi.getFilterName(), fmi.getMapping(), fmi.getDispatcher());
+                break;
+            default:
+                throw new IllegalStateException(
+                        "Unexpected " + io.undertow.servlet.api.FilterMappingInfo.MappingType.class.getName() + " "
+                                + fmi.getMappingType());
+            }
+        }
+        info.addListeners(src.getListeners());
+        info.addServletContainerInitalizers(src.getServletContainerInitializers());
+        for (ThreadSetupHandler a : src.getThreadSetupActions()) {
+            info.addThreadSetupAction(a);
+        }
+        for (Entry<String, String> en : src.getInitParameters().entrySet()) {
+            info.addInitParameter(en.getKey(), en.getValue());
+        }
+        for (Entry<String, Object> en : src.getServletContextAttributes().entrySet()) {
+            info.addServletContextAttribute(en.getKey(), en.getValue());
+        }
+        info.addWelcomePages(src.getWelcomePages());
```

Comment:
I am open to discuss the necessity of cloning individual properties. Currently the strategy is to clone everything and adapt only things known to require adjustments. Once https://github.com/undertow-io/undertow/pull/642 reaches us, we can drop the clone part of `adaptDeploymentInfo()`.

Reply:
Agreed. IMO we should strip it back to only the bits that we actually require.

Reply:
I understand your preference, but could you please explains the reasons behind that?

I have chosen the clone & adapt strategy because I believe it is better maintainable in the long term. Whatever aspect of deployment will be changed by WildFly in the future, we'll either get it for free or see that our tests fail. 

I also like the predictability of the approach. Given that the endpoint deloyment is cloned from the main deployment, its behavior is mostly given by the configuration of the main deployment.

Why should we remove just the welcome pages and error pages? Maybe the user wants them.

Reply:
Ok, fair point. Let's leave as-is for now.